### PR TITLE
Enable to give Image channels as different files

### PR DESCRIPTION
**Pull request by GFabien**
Make `Image` accept a sequence of paths and concatenate loaded tensors on the channel dimension. See https://github.com/fepegar/torchio/issues/253#issuecomment-669793181.

### CODE REVIEW

File: torchio/data/image.py [hunk 1]
```diff
@@ -124,7 +126,7 @@ class Image(dict):
     """
     def __init__(
             self,
-            path: Optional[TypePath] = None,
+            path: Union[TypePath, Sequence[TypePath]] = None,
```
Comment (fepegar) [inline torchio/data/image.py:129]: If `path` can be `None`, this should be `Optional[TypePath, Union[TypePath, Sequence[TypePath]]]` or `Union[TypePath, Sequence[TypePath], None]`. I think the latter is more readable.

File: torchio/data/image.py [hunk 2]
```diff
@@ -262,16 +264,35 @@ def get_bounds(self):
...
+                path = Path(path).expanduser()
+            except TypeError:
+                message = f'Conversion to path not possible for variable: ' \
+                          f'{path}'
+                raise TypeError(message)
+        else:
+            try:
+                path = [Path(p).expanduser() for p in path]
+            except TypeError:
+                message = f'Conversion to list of paths not possible for ' \
```
Comment (fepegar) [inline torchio/data/image.py:284]: Can you use brackets for consistency?

```python
message = (
    'sdfs'
    f'sdfl{sd}'
)
```

File: torchio/data/image.py [hunk 3]
```diff
@@ -262,16 +264,35 @@ def get_bounds(self):
...
+                          f'variable: {path}'
+                raise TypeError(message)
+
+        if isinstance(path, Path):
```
Comment (fepegar) [inline torchio/data/image.py:288]: Can the input be a list of dirs?

File: torchio/data/image.py [hunk 4]
```diff
@@ -309,11 +330,35 @@ def _load(self) -> Tuple[torch.Tensor, np.ndarray]:
...
+                warnings.warn(message, RuntimeWarning)
+
+            if not tensor.shape[1:] == new_tensor.shape[1:]:
+                message = f'Files shape do not match, found {tensor.shape}' \
+                          f'and {new_tensor.shape}'
+                RuntimeError(message)
+
+            tensors.append(new_tensor)
+
+        tensor = torch.cat(tensors, dim=0)
```
Comment (fepegar) [inline torchio/data/image.py:360]: I think you can omit `dim=0`.

File: torchio/utils.py [hunk 5]
```diff
@@ -43,12 +43,16 @@ def to_tuple(
...
+        ) -> Union[str, List[str]]:
     """
     '/home/user/image.nii.gz' -> 'image'
     """
-    path = Path(path)
-    return path.name.split('.')[0]
+    if isinstance(path, (str, Path)):
+        path = Path(path)
+        return path.name.split('.')[0]
+    return [Path(p).name.split('.')[0] for p in path]
```
Comment (fepegar) [inline torchio/utils.py:55]: Is there a way to not repeat this code? Maybe leave this as it was and write a new `get_stems`? Or an inner function `_get_stem` that returns `path.name.split('.')[0]`.

File: torchio/data/image.py (hunk 3 above)
Comment (GFabien) [reply torchio/data/image.py:288]: I don't know, it seems that for a single path a dir was acceptable, why not for a list of paths? Even if it corresponds to images with several channels it will work as long as they have the same spatial dimensions.

File: torchio/data/image.py (hunk 4 above)
Comment (GFabien) [reply torchio/data/image.py:360]: Maybe, but isn't it better if it's explicit?

File: torchio/data/image.py (hunk 3 above)
Comment (fepegar) [reply torchio/data/image.py:291]: It's just the impression I got by skimming through the code here. I'll get a better sense of it once I've properly gone through it.

File: torchio/data/image.py (hunk 4 above)
Comment (fepegar) [reply torchio/data/image.py:365]: I don't think it helps in this case, as it is the default. It's like using `print(s, sep='\n')` instead of `print(s)`.

File: tests/data/test_image.py [hunk 6]
```diff
@@ -17,6 +17,10 @@ def test_image_not_found(self):
         with self.assertRaises(FileNotFoundError):
             ScalarImage('nopath')
 
+    def test_wrong_path_value(self):
+        with self.assertRaises(TypeError):
+            ScalarImage('~&./@#"!?X7=+')
```
Comment (fepegar) [inline tests/data/test_image.py:22]: Looks like Obélix swearing haha.

However I'm not sure this should raise a `TypeError`. The `type` is `str`, which is fine. Why not `ValueError`?

Comment (GFabien) [reply tests/data/test_image.py:22]: Yes I got inspired for this one haha.

The caught one is a `RuntimeError`, do you prefer to have a `ValueError` or a `RuntimeError` then?

Comment (fepegar) [reply tests/data/test_image.py:22]: I think propagating that `RuntimeError` is the wise choice.

**Comment by fepegar** [comment]
Thanks, @GFabien! Added in `v0.17.26`.
